packages/strapi-plugin-users-permissions/controllers/User.js

```diff
@@ -170,23 +170,37 @@ module.exports = {
     const { id } = ctx.params;
     const { email, username, password } = ctx.request.body;
 
-    const userWithSameUsername = await strapi
-      .query('user', 'users-permissions')
-      .findOne({ username });
+    if(ctx.request.body.hasOwnProperty('username') && !username) {
```

Comment:
Hey can you please use _.has from lodash instead

Reply:
Apologies did not have the time yet! Looks like #4040 was one step ahead 😅

Reply:
No roblem thanks again for you help !